Assets/Scripts/ObsticleSpawner.cs

```diff
@@ -2,23 +2,70 @@
 using System.Collections.Generic;
 using UnityEngine;
 
-public class ObsticleSpawner : MonoBehaviour
+namespace AYellowpaper.SerializedCollections
 {
-    public GameObject obsticle;
+	public class ObsticleSpawner : MonoBehaviour
+	{
+		[SerializedDictionary("Interactable", "Weight")]
+		public SerializedDictionary<InteractableType, int> InteractableWeights;
+		public SerializedDictionary<InteractableType, GameObject> InteractableObjects;
```

Comment:
since no other class should be able to edit these dicts, make it private with [SerializedField]